homeassistant/components/tplink/light.py

```diff
@@ -55,12 +55,12 @@
     return True
 
 
-def add_entity(device: SmartBulb, async_add_entities):
+async def add_entity(device: SmartBulb, async_add_entities):
     """Check if device is online and add the entity."""
     # Attempt to get the sysinfo. If it fails, it will raise an
     # exception that is caught by async_add_entities_retry which
     # will try again later.
-    device.get_sysinfo()
+    await device.update()
```

Comment:
Here you're updating it again, but you also just updated it during discovery ?

Reply:
I tried to keep the existing semantics in place for the initial testing. The reasoning for this update was to trigger an exception during the initial setup (maybe the device was manually configured, but not available during the startup?).